src/widgets/allergies/allergies-overview.component.tsx

```diff
@@ -71,22 +77,31 @@ const AllergiesOverview: React.FC<AllergiesOverviewProps> = () => {
     openWorkspaceTab(AllergyForm, t("allergiesForm", "Allergies Form"));
   };
 
+  const handlePageChange = ({ page }) => {
+    setPageNumber(page);
+  };
+
+  React.useEffect(() => {
+    if (!isEmpty(allergies)) {
```

Comment:
Hm to me all of that rather sounds like a use of `useMemo` than `useEffect`. You can just *always* compute the value for the current page entries, so `useMemo` may be great since it improves perf.

Reply:
i think if this were using use memo, it would be implemented like so.

```
const [page, allPages] = React.useMemo(() => {
   return paginate<any>(allergies, pageNumber, pageSize);
}, [allergies, pageNumber, pageSize]]);
```
if you do it this way, you probably don't need initialize state for page / currentpage, right?